portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,9 @@
      factContainer.innerText = randomFact;
  }
 
+//Fetches a message from the server and adds it to the DOM
+function getFetchedMessage(){
+    fetch('/data').then(response => response.text()).then ((message) => {
+      document.getElementById('message-container').innerText = message;
+    });
```

Comment:
Also add error handling in case the call fails.

Reply:
Just add
.catch((error) => {
  console.error('Error:', error);
});

Reply:
sounds good, thanks!